examples/dao/assets/proposal-lsig.py

```diff
@@ -0,0 +1,56 @@
+import sys
+sys.path.insert(0,'..')
+
+from algobpy.parse import parse_params
+from pyteal import *
+
+def proposal_lsig(ARG_OWNER):
+    """
+    Represents Proposal Lsig (a contract account owned by the proposer)
+    """
+
+    # check no rekeying, close remainder to, asset close to for a txn
+    def basic_owner_checks(txn: Txn): return And(
+        txn.rekey_to() == Global.zero_address(),
+
+        # only owner can close ASA/ALGO
+        txn.sender() == Addr(ARG_OWNER)
+    )
+
+    def basic_checks(txn: Txn): return And(
+        txn.rekey_to() == Global.zero_address(),
+        txn.close_remainder_to() == Global.zero_address(),
+        txn.asset_close_to() == Global.zero_address()
+    )
+
+    # Opt-in transaction.
+    opt_in = And(
+        basic_owner_checks(Txn),
```

Comment:
i think while opting in `txn.sender` will be proposal lsig address, not owner address. so this check may fail here. could yo u please check ?

Reply:
yes, removed

Reply:
but we need to ensure opt-in can only be done by owner, so we can ensure that by checking if transaction is signed by owner.

Reply:
ok, adding check for sender